pkg/tests/validator_test.go

```diff
@@ -0,0 +1,381 @@
+package tests
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/iotaledger/hive.go/core/safemath"
+	"github.com/iotaledger/hive.go/lo"
+	"github.com/iotaledger/hive.go/log"
+	"github.com/iotaledger/hive.go/runtime/options"
+	"github.com/iotaledger/iota-core/pkg/testsuite"
+	"github.com/iotaledger/iota-core/pkg/testsuite/mock"
+	"github.com/stretchr/testify/require"
+
+	iotago "github.com/iotaledger/iota.go/v4"
+)
+
+// IOTA Mainnet Max Supply.
+const MAX_SUPPLY = iotago.BaseToken(4_600_000_000_000_000)
+
+func setupValidatorTestsuite(t *testing.T, walletOpts ...options.Option[testsuite.WalletOptions]) *testsuite.TestSuite {
+	var slotDuration uint8 = 5
+	var slotsPerEpochExponent uint8 = 5
+	var validationBlocksPerSlot uint8 = 5
+
+	ts := testsuite.NewTestSuite(t,
+		testsuite.WithProtocolParametersOptions(
+			iotago.WithSupplyOptions(MAX_SUPPLY, 63, 1, 17, 32, 21, 70),
+			iotago.WithStakingOptions(1, validationBlocksPerSlot, 1),
+			// Pick larger values for ManaShareCoefficient and DecayBalancingConstant for more precision in the calculations.
+			iotago.WithRewardsOptions(8, 8, 11, 200, 200),
+			// Pick Increase/Decrease threshold in accordance with sanity checks (necessary because we changed slot duration).
+			iotago.WithCongestionControlOptions(1, 0, 0, 400_000, 300_000, 100_000, 1000, 100),
+			iotago.WithTimeProviderOptions(
+				0,
+				testsuite.GenesisTimeWithOffsetBySlots(1000, slotDuration),
+				slotDuration,
+				slotsPerEpochExponent,
+			),
+			iotago.WithLivenessOptions(
+				10,
+				20,
+				5,
+				10,
+				15,
+			),
+		),
+	)
+
+	// Add validator nodes to the network. This will add validator accounts to the snapshot.
+	vnode1 := ts.AddValidatorNode("node1", append(
+		[]options.Option[testsuite.WalletOptions]{
+			testsuite.WithWalletAmount(20_000_000),
+		},
+		walletOpts...,
+	)...)
+	vnode2 := ts.AddValidatorNode("node2", append(
+		[]options.Option[testsuite.WalletOptions]{
+			testsuite.WithWalletAmount(25_000_000),
+		},
+		walletOpts...,
+	)...)
+
+	// Add a non-validator node to the network. This will not add any accounts to the snapshot.
+	node3 := ts.AddNode("node3")
+	// Add a default block issuer to the network. This will add another block issuer account to the snapshot.
+	ts.AddDefaultWallet(vnode1)
+
+	ts.Run(true)
+
+	// For debugging set the log level appropriately.
+	vnode1.Protocol.SetLogLevel(log.LevelInfo)
+	vnode2.Protocol.SetLogLevel(log.LevelError)
+	node3.Protocol.SetLogLevel(log.LevelError)
+
+	ts.SetCurrentSlot(1)
```

Comment:
Should we actually set the default CurrentSlot to be 1 when we create a new testsuite? Or rather set it to GenesisSlot + 1?